File: biblioteca-web/src/app/models/waitlist.model.ts
export interface WaitlistEntry {
  id: number;
  bookId: number;
  bookTitle: string;
  studentName: string;
  matricula: string;
  requestDate: string;
  position: number;
  status:
    | 'esperando'
    | 'notificado'
    | 'reserva_confirmada'
    | 'vencido'
    | 'cancelado';
  reservedUntil?: number;
}

File: backend/biblioteca-api/app/models/waitlist_model.py
from typing import Literal, Optional
from pydantic import BaseModel


class WaitlistCreate(BaseModel):
    id: int
    bookId: int
    bookTitle: str
    studentName: str
    matricula: str
    requestDate: str
    position: int

    status: Literal[
        "esperando",
        "notificado",
        "reserva_confirmada",
        "vencido",
        "cancelado"
    ] = "esperando"

    reservedUntil: Optional[int] = None